app/core/auth-options.ts

```diff
@@ -233,23 +302,39 @@ export const authOptions: AuthOptions = {
       return true;
     },
     async jwt({ token, account }: { token: any; account: Account | null }) {
-      if (account?.access_token) {
-        const decodedToken: any = jwt.decode(account.access_token);
-
+      const now = new Date();
+      let user;
+      if (account) {
         token.accessToken = account.access_token;
+        token.refreshToken = account.refresh_token;
         token.idToken = account.id_token;
-        token.roles = _get(decodedToken, `resource_access.${AUTH_RESOURCE}.roles`, []);
-        token.sub = decodedToken?.sub ?? '';
-        token.teams = [];
-
-        _forEach(decodedToken.resource_access ?? {}, (val, key) => {
-          if (key.startsWith(TEAM_SA_PREFIX)) {
-            token.teams.push({ clientId: key, roles: val.roles });
-          }
+        Object.assign(token, await processToken(token));
+      }
+
+      try {
+        user = await prisma.user.update({
+          where: {
+            email: token.email,
+            nextTokenRefreshTime: { lt: now },
+          },
+          data: {
+            nextTokenRefreshTime: addMinutes(now, USER_TOKEN_REFRESH_MIN),
+          },
         });
+      } catch (error) {}
+
+      if (user) {
+        try {
+          const { access_token } = await getNewTokens(token.refreshToken);
+          token.accessToken = access_token;
```

Comment:
what about other tokens?
otherwise, how do we update the newest `refresh token` every time?